### PR TITLE
Feature/menu

### PR DESCRIPTION
**Pull request by vaisami**

### CODE REVIEW

File: examples/app.component.jsx [hunk 1]
```diff
@@ -20,8 +21,28 @@ const Header = () => {
   return content;
 };
 
+const getPrefix = (item) => {
+  const content = null;
+  //content = (<Icon type="product" name="Invoices" height={25} width={25} />);
```
Comment (ilkkalehtinen) [inline examples/app.component.jsx]: Remove commented code

File: examples/components/menu-view/menu-view.component.jsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { Menu } from '../../../src/index';
+
+function range(start, end) {
+  const list = [];
+  /* eslint-disable no-plusplus  */
```
Comment (ilkkalehtinen) [inline examples/components/menu-view/menu-view.component.jsx]: Fix these properly ( i += 1).

File: examples/components/menu-view/menu-view.component.jsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  for (let i = start; i <= end; i++) {
+    list.push(i);
+  }
+  return list;
+}
+
+const getItems = () => {
+  const label = 'Menu Item';
+
+  /* eslint-disable arrow-body-style */
```
Comment (ilkkalehtinen) [inline examples/components/menu-view/menu-view.component.jsx]: Same here.

File: examples/layout/menu.constants.js [hunk 4]
```diff
@@ -1,82 +1,90 @@
...
-  //   to: '/introduction',
-  // },
-  // {
-  //   name: 'Style',
-  //   to: '/style',
-  // },
-  // {
-  //   name: 'Technology',
-  //   to: '/technology',
+  //   id: 100,
```
Comment (ilkkalehtinen) [inline examples/layout/menu.constants.js]: No commented code.

File: src/menu/key-codes.constant.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+const KEY_CODES = {
```
Comment (ilkkalehtinen) [inline src/menu/key-codes.constant.js]: Should there be a common place for these as this is not just 'menu' specific?

Comment (vaisami) [inline src/menu/key-codes.constant.js]: Yes

Comment (vaisami) [inline src/menu/key-codes.constant.js]: src/constants for the time being at least.

**Pull request merged**